internal/util/sizeof_test.go

```diff
@@ -0,0 +1,376 @@
+
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/gcs"
+	. "github.com/jacobsa/ogletest"
+	"google.golang.org/api/googleapi"
+	storagev1 "google.golang.org/api/storage/v1"
+)
+
+func TestSizeof(t *testing.T) { RunTests(t) }
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+type SizeofTest struct {
+}
+
+func init() { RegisterTestSuite(&SizeofTest{}) }
+
+var (
+	i            int
+	intArray     []int
+	b            byte
+	emptyString  string = ""
+	stringArray  []string
+	stringIntMap map[string]int
+
+	sizeOfInt               int
+	sizeOfIntPtr            int
+	sizeOfByte              int
+	sizeOfEmptyString       int
+	sizeOfEmptyIntArray     int
+	sizeOfEmptyStringArray  int
+	sizeOfEmptyStringIntMap int
+	sizeOfEmptyStruct       int
+	sizeOfEmptyGcsObject    int
+	sizeOfEmptyMinObject    int
+)
+
+func init() {
+	type emptyStruct struct{}
+
+	sizeOfInt = int(unsafe.Sizeof(i))
+	sizeOfIntPtr = int(unsafe.Sizeof(&i))
+	sizeOfByte = int(unsafe.Sizeof(b))
+	sizeOfEmptyString = int(unsafe.Sizeof(emptyString))
+	sizeOfEmptyIntArray = int(unsafe.Sizeof(intArray))
+	sizeOfEmptyStringArray = int(unsafe.Sizeof(stringArray))
+	sizeOfEmptyStringIntMap = int(unsafe.Sizeof(stringIntMap))
+
+	sizeOfEmptyStruct = int(unsafe.Sizeof(emptyStruct{}))
+	AssertEq(0, sizeOfEmptyStruct)
+
+	sizeOfEmptyGcsObject = int(unsafe.Sizeof(gcs.Object{}))
+	sizeOfEmptyMinObject = int(unsafe.Sizeof(gcs.MinObject{}))
+}
+
+////////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////////
+
+func (t *SizeofTest) TestUnsafeSizeOf() {
+
+	for _, tc := range []struct {
+		t             any
+		expected_size int
+	}{
+		{
+			t:             i,
+			expected_size: sizeOfInt,
+		},
+		{
+			t:             &i,
+			expected_size: sizeOfIntPtr,
+		},
+		{
+			t: struct {
+			}{},
+			expected_size: sizeOfEmptyStruct,
+		},
+		{
+			t: struct {
+				x int
+			}{},
+			expected_size: sizeOfEmptyStruct + sizeOfInt,
+		},
+		{
+			t: struct {
+				a          int
+				b1, b2, b3 byte
+				c          string
+			}{},
+			expected_size: sizeOfEmptyStruct + sizeOfInt + 3*sizeOfByte + 5 /*for-padding-for-alignment*/ + sizeOfEmptyString,
+		},
+		{
+			t:             "",
+			expected_size: sizeOfEmptyString,
+		},
+		{
+			t:             "hello",
+			expected_size: sizeOfEmptyString,
+		},
+		{
+			t:             []int{1, 2, 3},
+			expected_size: sizeOfEmptyIntArray,
+		},
+		{
+			t:             []string{"few ", "fewfgwe", "", "fewawef"},
+			expected_size: sizeOfEmptyStringArray,
+		},
+		{
+			t:             map[string]int{"few ": 432, "fewfgwe": -21, "": 1, "fewawef": 0},
+			expected_size: sizeOfEmptyStringIntMap,
+		},
+	} {
+		calculatedSize := UnsafeSizeOf(&tc.t)
+		AssertEq(tc.expected_size, calculatedSize)
+	}
+}
+
+func (t *SizeofTest) TestContentSizeOfString() {
```

Comment:
Since, we are testing both methods, contentSizeOfString and nestedSizeOfString.

So, we can rename:
Test_ContentSizeOfString_And_NestedCotentSizeOfString() {
}

and table struct will contain three entries - string, expected_content_size and expected_nested_content_size.

Reply:
Renamed. The reason I didn't keep expected_nested_size separately as it's always expected to be = sizeOfEmptyString + expected_content_size, so duplication of code. Anyway, I've added it now.